supertokens/utils_test.go

```diff
@@ -54,3 +56,44 @@ func TestGetTopLevelDomainForSameSiteResolution(t *testing.T) {
 		assert.Equal(t, val.Output, domain, val.Input)
 	}
 }
+
```

Comment:
ideally we want to add a test in which you send a full request to the middleware (like sign in APi call) and then check that the logger outputs with the request ID correctly when:
- request ID is a part of the header
- request Id is a part of the context with a specific requestIdKey
- When neither of those are given, the request ID is not in the logging

Reply:
Added tests